global_install/hooks/pre-commit

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
```

Comment:
This file is mostly  a copy of the `precommit framework` file. Don't @-me about the bad python. yet.

Reply:
Having read more of the python, it's pretty good, for cross-platform resilience.